vomsimporter.py

```diff
@@ -571,16 +572,45 @@ def import_voms_user(self, voms_user):
                     voms_user, iam_user, cern_login)
 
     def resolve_cern_login_from_ldap(self, voms_user):
-        cern_login = subprocess.check_output(["resolve_cern_login", str(voms_user[
-            'cernHrId']), self._ldap_host, self._ldap_port]).replace("\n", "").strip()
+        cern_login = None
+        lfilter = "(&(objectClass=user)(employeeType=Primary)(employeeID={0}))".format(voms_user['cernHrId'])
 
-        if len(cern_login) == 0:
-            logging.warn("CERN login resolution failed for personId %s", voms_user[
-                'cernHrId'])
-            return None
+        ldap.set_option(ldap.OPT_REFERRALS, 0)
+        l = ldap.initialize("ldap://{0}:{1}".format(self._ldap_host, self._ldap_port))
+
+        try:
+            #sasl_auth = ldap.sasl.sasl({},'GSSAPI')
+            #l.sasl_interactive_bind_s("", sasl_auth)
+            l.simple_bind_s('','')
+
+            r = l.search_s("DC=cern,DC=ch", ldap.SCOPE_SUBTREE, lfilter, [ 'cn', 'proxyAddresses' ])
+            if len(r) == 0:
+                logging.warn("CERN login resolution failed for personId %s",
+                    voms_user['cernHrId'])
+                return None
+
+            dn, attrs = r[0]
```

Comment:
Would it make sense to check also for a length greater than one?

Reply:
There is exactly one record for `employeeType=Primary` for existing HR ID.